File: 2015/src/day01/day01.ts
import { Task } from "../task";

export class Day01 extends Task {
  partOne(): String {
    let input = this.readInput();
    let instructions = input.split("");

    const map: Map<string, number> = new Map();

    instructions.forEach((element) => {
      let item = map.get(element);

      if (item === undefined) item = 0;

      map.set(element, element === "(" ? item + 1 : item - 1);
    });

    let floor = Array.from(map.values()).reduce((acc, val) => acc + val);
    return `${floor}`;
  }
  partTwo(): String {
    let input = this.readInput();
    let instructions = input.split("");

    let floor = 0;
    let position = 0;
    for (let element of instructions) {
      if (element === "(") {
        floor++;
      } else {
        floor--;
      }
      position++;

      if (floor === -1) break;
    }

    return `${position}`;
  }
}

File: 2024/day01/Day01.cs
namespace AdventOfCode.day01
{
    internal class Day01 : DailyTask
    {
        public override string PartOne()
        {
            var firstList = new List<int>();
            var secondList = new List<int>();

            var diffs = new List<int>();
            var splittedInput = Input.Split(Environment.NewLine);
            foreach (var input in splittedInput)
            {
                var items = input.Split("   ");
                firstList.Add(int.Parse(items[0]));
                secondList.Add(int.Parse(items[1]));
            }
            firstList.Sort();
            secondList.Sort();

            foreach (var nw in firstList.Zip(secondList, Tuple.Create))
            {
                diffs.Add(Math.Abs(nw.Item1 - nw.Item2));
            }

            return diffs.Sum().ToString();
        }

        public override string PartTwo()
        {
            var firstList = new List<int>();
            var secondList = new List<int>();

            var nums = new List<int>();

            var diffs = new List<int>();
            var splittedInput = Input.Split(Environment.NewLine);
            foreach (var input in splittedInput)
            {
                var items = input.Split("   ");
                firstList.Add(int.Parse(items[0]));
                secondList.Add(int.Parse(items[1]));
            }

            foreach (var item in firstList)
            {
                var hits = secondList.FindAll(x => x == item);
                nums.Add(item * hits.Count);
            }

            return nums.Sum().ToString();
        }
    }
}
